src/client/components/pages/RoomAdmin/__tests__/RoomAdminTable.test.tsx

```diff
@@ -21,6 +24,186 @@ import {
+      bodyRow.forEach((row) => {
+        strictEqual(
+          within(row).queryByText('Bauer Laboratory') !== null,
+          true,
+          'Bauer Laboratory is not in the table as expected'
+        );
+      });
+    });
+  });
+});
+describe('Room input filter', function () {
+  let getStub: SinonStub;
+  let renderResult: RenderResult;
+  beforeEach(function () {
+    getStub = stub(LocationAPI, 'getAdminRooms');
+    getStub.resolves(adminRoomsResponse);
+    renderResult = render(
+      <RoomAdmin />
+    );
+  });
+  it('defaults to null', async function () {
+    await wait(() => renderResult.getAllByRole('row').length > 1);
+    const roomFilterInput = renderResult.getByLabelText('The table will be filtered by selected room name', { exact: true }) as HTMLSelectElement;
+    strictEqual(roomFilterInput.value, '');
+  });
+  it('filters table data when room name is entered', async function () {
+    await wait(() => renderResult.getAllByRole('row').length > 1);
+    const [roomInput] = renderResult.getAllByLabelText('The table will be filtered by selected room name');
+    fireEvent.change(roomInput, { target: { value: '101A' } });
+    await wait(() => renderResult.getAllByRole('row').length > 1);
+    const rows = renderResult.getAllByRole('row');
+    const bodyRow = rows.filter((row) => (
+      within(row).queryAllByRole('columnheader').length === 0
+    ));
+    bodyRow.forEach((row) => {
+      strictEqual(
+        within(row).queryByText('101A') !== null,
+        true,
+        '101A is not in the table as expected'
+      );
+    });
+  });
+});
+describe('capacity input filter', function () {
+  let getStub: SinonStub;
+  let renderResult: RenderResult;
+  beforeEach(function () {
+    getStub = stub(LocationAPI, 'getAdminRooms');
+    getStub.resolves(adminRoomsResponse);
+    renderResult = render(
+      <RoomAdmin />
+    );
+  });
+  it('defaults to null', async function () {
+    await wait(() => renderResult.getAllByRole('row').length > 1);
+    const capacityFilterInput = renderResult.getByLabelText('The table will be filtered by entered capacity', { exact: true }) as HTMLSelectElement;
+    strictEqual(capacityFilterInput.value, '');
+  });
+  it('filters table data by entered capacity entered', async function () {
+    await wait(() => renderResult.getAllByRole('row').length > 1);
+    const [capacityFilterInput] = renderResult.getAllByLabelText('The table will be filtered by selected room name');
+    fireEvent.change(capacityFilterInput, { target: { value: '20' } });
+    await wait(() => renderResult.getAllByRole('row').length > 1);
+    const rows = renderResult.getAllByRole('row');
+    const bodyRow = rows.filter((row) => (
+      within(row).queryAllByRole('columnheader').length === 0
+    ));
+    bodyRow.forEach((row) => {
+      strictEqual(
+        within(row).queryByText('20') !== null,
+        true,
+        'room capacity filter is not populating data properly'
+      );
+    });
+  });
```

Comment:
I would also define a `testCapacity` variable at the top of this test, since '20' is being referenced in more than one place. This is a good practice so that if you ever need to change the value from '20', you just have to make the variable a different value rather than changing the '20' value in multiple places. I know it's only being used twice here, but I think it's a good practice in general nonetheless.